elements/lisk-tree/src/sparse_merkle_tree/sparse_merkle_tree.ts

```diff
@@ -35,8 +42,114 @@ export class SparseMerkleTree {
 	public get db(): Database {
 		return this._db;
 	}
+
+	public async getNode(nodeHash: Buffer): Promise<TreeNode> {
+		if (nodeHash.equals(EMPTY_HASH)) {
+			return new Empty();
+		}
+		const data = await this._db.get(nodeHash);
+
+		if (!data) {
+			throw new Error(
+				`Node with input hash: ${nodeHash.toString('hex')} does not exist in the tree`,
+			);
+		}
+		if (isLeaf(data)) {
+			const { key, value } = parseLeaf(data, this.keyLength);
+
+			return new Leaf(key, value);
+		}
+
+		const { leftHash, rightHash } = parseBranch(data);
+
+		return new Branch(leftHash, rightHash);
+	}
+	// As specified in from https://github.com/LiskHQ/lips/blob/master/proposals/lip-0039.md
+	public async update(key: Buffer, value: Buffer) {
```

Comment:
Is it not necessary to declare the return type ?
```suggestion
	public async update(key: Buffer, value: Buffer): Promise<TreeNode> {
```